SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -0,0 +1,806 @@
+        SDLTransportSelectionBox *transportSelection = [secondaryTransports objectAtIndex:0];
+        secondaryTransportSelection = [transportSelection integerValue];
+    }
+
+    SDLTransportSelection primaryTransportSelection = [self sdl_getTransportSelectionFromProtocol:self.primaryProtocol];
+    if (primaryTransportSelection == secondaryTransportSelection) {
+        SDLLogW(@"Same transport is specified for both primary and secondary transport. Secondary transport is disabled.");
+        secondaryTransportSelection = SDLTransportSelectionDisabled;
+        // clear out these values, so that audio and video services will start on primary transport
+        transportsForAudio = nil;
+        transportsForVideo = nil;
+    } else if (secondaryTransportSelection == SDLTransportSelectionIAP) {
+        SDLLogW(@"Starting IAP as secondary transport, which does not usually happen");
+    }
+
+    self.transportSelection = secondaryTransportSelection;
+    self.transportsForAudioService = transportsForAudio;
+    self.transportsForVideoService = transportsForVideo;
+
+    // this will trigger audio / video streaming if they are allowed on primary transport
+    [self sdl_handleTransportUpdate:YES secondaryAvailable:NO];
+
+    [self.stateMachine transitionToState:SDLSecondaryTransportStateConfigured];
+}
+
+- (void)sdl_handleTransportEventUpdate {
+    if ([self.stateMachine isCurrentState:SDLSecondaryTransportStateStarted]) {
+        // The system sent Transport Event Update frame prior to Start Service ACK! Just keep the information and do nothing here.
+        SDLLogD(@"Received TCP transport information prior to Start Service ACK");
+        return;
+    }
+    if (self.transportSelection != SDLTransportSelectionTCP) {
+        return;
+    }
+
+    if ([self.stateMachine isCurrentState:SDLSecondaryTransportStateConfigured] && [self sdl_isTCPReady]) {
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateConnecting];
+    } else if ([self sdl_isTransportOpened]) {
+        // Disconnect current transport. If the IP address is available then we will reconnect immediately.
+        SDLLogD(@"TCP transport information updated, disconnecting current transport");
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateConfigured];
+    } else if ([self.stateMachine isCurrentState:SDLSecondaryTransportStateReconnecting]) {
+        SDLLogD(@"TCP transport information updated, aborting reconnection timer");
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateConfigured];
+    }
+}
+
+- (void)sdl_handleTransportRegistered {
+    // secondary transport is now ready
+    [self.stateMachine transitionToState:SDLSecondaryTransportStateRegistered];
+}
+
+- (void)sdl_handleTransportRegisterFailed {
+    [self.stateMachine transitionToState:SDLSecondaryTransportStateReconnecting];
+}
+
+- (void)sdl_handleTransportRegisterTimeout {
```

Comment:
This one-liner is only called from one place, just move the code there?

Reply:
Updated in commit b734d113d4918e8fe6af7ed6c303c017d0582083.